### PR TITLE
Fix template_fields type to have MyPy friendly Sequence type

### PR DESCRIPTION
**Pull request by potiuk**
Part of #19891

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Another "global" mypy fix - constistently set type of "template_fiellds" (fixes quite a number of mypy bugs across many packages).

File: airflow/providers/amazon/aws/operators/eks.py [hunk 1]
```diff
@@ -623,7 +623,7 @@ class EksPodOperator(KubernetesPodOperator):
     :type aws_conn_id: str
     """
 
-    template_fields: Iterable[str] = {
+    template_fields: Sequence[str] = {
```
Comment (josh-fell) [inline airflow/providers/amazon/aws/operators/eks.py]: Looks like the expression is a set here. Convert to a tuple maybe?

File: airflow/providers/qubole/operators/qubole_check.py [hunk 2]
```diff
@@ -105,7 +105,7 @@ class QuboleCheckOperator(_QuboleCheckOperatorMixin, SQLCheckOperator, QuboleOpe
 
     """
 
-    template_fields: Iterable[str] = set(QuboleOperator.template_fields) | set(
+    template_fields: Sequence[str] = set(QuboleOperator.template_fields) | set(
```
Comment (josh-fell) [inline airflow/providers/qubole/operators/qubole_check.py]: Expression is a set here too.

File: airflow/providers/amazon/aws/operators/eks.py (hunk 1 above)
Comment (potiuk) [reply airflow/providers/amazon/aws/operators/eks.py]: Why not.

File: airflow/providers/qubole/operators/qubole_check.py (hunk 2 above)
Comment (potiuk) [reply airflow/providers/qubole/operators/qubole_check.py]: Yeah. Good points @josh-fell  - I will replace all uses with tuples (including those that are using lists).

**Review by dstandish — APPROVED** [review]
looked over all files

**Comment by potiuk** [comment]
I replaced all the arrays and set with tuples too.

**Review by mik-laj — APPROVED** [review]

**Pull request merged**